[FEAT] 채팅방 입장 시 WebSocket 연결 후 채팅 전송 기능 구현

dye0p opened the pull request:
## Issue Number
closed #812

## As-Is
<!-- 문제 상황 정의 -->
실시간 양방향 통신을 위한 웹소켓 채팅 기능이 구현되어 있지 않아, 사용자들이 실시간으로 메시지를 주고받을 수 없습니다.

## To-Be
<!-- 변경 사항 -->
- Spring WebSocket과 STOMP 프로토콜을 기반으로 실시간 채팅 기능을 구현했습니다.
- 웹소켓 설정 추가:
  - STOMP 엔드포인트(/ws-chat) 및 메시지 브로커(/topic)를 설정했습니다.
  - 웹소켓 연결 시 JWT 토큰을 검증하는 AuthHandshakeInterceptor를 추가하여 보안을 강화했습니다.
- 채팅 메시지 처리 로직 구현:
  - @MessageMapping을 사용하는 ChatController를 추가하여 클라이언트의 메시지 발행을 처리합니다.
  - ChatMessageService를 통해 수신된 메시지를 데이터베이스에 저장하고, 해당 채팅방 구독자들에게 브로드캐스팅합니다.
 데이터베이스 스키마 추가:
  - 채팅 메시지를 저장하기 위한 chat_message 테이블 생성 스크립트를 추가했습니다.(V202510021642__create_chat_message_schema.sql)
- 로그인 응답 구조 변경:
  - 로그인 시 반환되는 memberId를 MemberLoginResponse DTO로 감싸도록 응답 구조를 수정했습니다.


## Check List
- [x] 테스트가 전부 통과되었나요?
- [x] 모든 commit이 push 되었나요?
- [x] merge할 branch를 확인했나요?
- [x] Assignee를 지정했나요?
- [x] Label을 지정했나요?
- [x] 닫을 이슈 번호를 지정했나요?


## (Optional) Additional Description
- 클라이언트는 /app/chatroom/{chatRoomId}로 메시지를 발행(send)하고, /topic/chatroom/{chatRoomId}를 구독(subscribe)하여 메시지를 수신할 수 있습니다.
- 웹소켓 연결(handshake) 과정에서 쿠키에 담긴 accessToken을 통해 사용자 인증이 이루어집니다. 인증에 실패하면 연결이 거부됩니다.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- 신기능
  - 웹소켓(STOMP) 기반 실시간 채팅 도입: 인증된 사용자만 참여, 채팅방별 토픽 구독/수신 지원.
  - 채팅 메시지 영속화 및 소프트 삭제 지원. 전송 시 tempId 매핑과 생성 시각 포함 응답.
- API 변경
  - 로그인 응답 구조 변경: memberId가 하위 객체로 래핑되어 반환됩니다(기존 필드 직접 노출 제거). 클라이언트 파싱 로직 수정 필요.
- 데이터베이스
  - 채팅 메시지 테이블 신규 생성(콘텐츠, 발신자, 채팅방, 생성/삭제 메타데이터 포함).
- 잡무
  - 웹소켓 지원 라이브러리 의존성 추가.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

dye0p committed:
feat: 채팅방 관련 예외 클래스 및 핸들러 추가

채팅방 기능 구현에 필요한 커스텀 예외 클래스를 추가했음. 각 상황에 맞는 처리를 위해 예외를 세분화할 필요가 있었기 때문.

- `ChatRoomAlreadyExistsException`: 채팅방이 이미 존재할 경우 발생

- `ChatRoomNotFoundException`: 채팅방을 찾을 수 없는 경우 발생

- `UnauthorizedChatRoomAccessException`: 채팅방 접근 권한이 없는 경우 발생

추가된 예외들을 처리하기 위해 `GlobalExceptionHandler`에 핸들러 메서드를 추가함.

dye0p committed:
feat: 채팅방 조회 기능 추가

- `ChatRoomService`에 채팅방 조회 메서드 추가 (`findChatRoom`)
- 채팅방 존재 여부 및 참여자 권한 검증 로직 구현
- 예약 상태 확인을 통해 접근 가능 여부 검증 (`isChatRoomAccessibleStatus`)
- 멤버 정보와 멘토링 정보로 `ChatRoomResponse` 생성
- `Member`, `Reservation`, `ChatRoom`에 필요한 유틸 메서드(`isMentee`, `hasParticipant`) 추가
- `ChatRoomResponse` 클래스에 정적 팩토리 메서드 및 프로필 이미지 URL 처리 로직 추가

dye0p committed:
feat: 채팅방 단건 조회 API 컨트롤러 추가

- `@GetMapping`으로 `/chatrooms/{chatroomId}` 엔드포인트 정의
- 채팅방 조회 시 인증 필요하도록 `@AuthRequired` 추가
- `ChatRoomService.findChatRoom` 호출로 채팅방 정보 조회 및 응답
- 로그인 정보(`LoginInfo.memberId`)와 PathVariable(`chatroomId`)로 조회
- 응답 데이터로 `ChatRoomResponse` 반환

dye0p committed:
feat: 채팅방 생성 로직 보완 및 예약 검증 로직 추가

- `ChatRoomService.registerChatRoom`에 예약 및 멘토링 검증 메서드 추가
- `validateMentoring`: 존재하지 않는 멘토링 예외 처리 추가
- `validateReservationExists`: 이미 생성된 채팅방인지 확인 후 예외 처리
- `ChatRoomRepository`에 `existsByReservationId` 메서드 추가 (예약 기반 중복 확인)

dye0p committed:
test: ChatRoomService 테스트 추가

- 채팅방 생성(registerChatRoom) 성공/실패 케이스 테스트 추가
- 멘티/멘토의 채팅방 조회 성공/실패 케이스 테스트 추가
- 예약 상태 및 소프트 삭제된 멘토링 검증 테스트 추가
- 프로필 이미지 유효성 확인 테스트 추가

dye0p committed:
test: 채팅방 접근 예외 메시지 수정 검증 테스트 추가

- UnauthorizedChatRoomAccessException 메시지를 INVALID_STATUS_CHAT_ROOM_ACCESS로 변경 후 검증
- BusinessErrorMessage 변경 사항에 따른 테스트 수정

dye0p committed:
refactor: 컨벤션 수정

dye0p committed:
build: 웹소켓 의존성 추가

dye0p committed:
refactor: 잘못된 패키지 위치 이동

dye0p committed:
refactor: 불필요한 final 키워드 제거

dye0p committed:
feat: 채팅방 엔티티 채팅 상태 필드 추가

- flyway chat_room 테이블 status 칼럼 추가 스크립트 추가 (VARCHAR, NOT NULL)
- 채팅방의 상태 관리 필요성을 느껴 필드 추가

dye0p committed:
feat: 로그인 응답에 회원 식별자 추가

- `LoginResponse` DTO 생성 (memberId와 AuthToken 포함)
- AuthService의 `login` 메서드 반환 타입 변경 (`AuthTokenResponse` → `LoginResponse`)
- AuthController의 `/login` API 응답 변경 (회원 식별자 반환 추가)
- 관련 테스트 수정 및 정상 응답 데이터 검증 항목 추가

dye0p committed:
feat: 로그인 응답에 회원 식별자 추가

- `LoginResponse` DTO 생성 (memberId와 AuthToken 포함)
- AuthService의 `login` 메서드 반환 타입 변경 (`AuthTokenResponse` → `LoginResponse`)
- AuthController의 `/login` API 응답 변경 (회원 식별자 반환 추가)
- 관련 테스트 수정 및 정상 응답 데이터 검증 항목 추가

dye0p committed:
refactor: ChatRoomResponse 응답 필드 수정

- 기존 ChatRoomResponse에 포함되어있던 멘토링 관련 필드를 제거하였음
- 멘토링의 정보는 별도의 API로 관리하는 것이 관심사를 분리하는 것에 적절하다 판단하였음
- "상대방"이라는 뜻을 나타내기 위해서 participant 네이밍을 사용했지만, opponent이 더 적절하다 판단하여 변경하였음

dye0p committed:
feat: 웹소켓 설정 추가

- WebSocketConfig 클래스 생성 및 @EnableWebSocketMessageBroker 적용
- 메시지 브로커 구성: /app으로 시작하는 메시지 처리 및 /topic 브로커 사용 설정
- Stomp 엔드포인트 /ws-chat 등록

dye0p committed:
feat: WebSocket STOMP 엔드포인트 설정 수정

- /ws-chat 엔드포인트에 setAllowedOriginPatterns("*") 추가
- SockJS 지원을 위해 withSockJS() 호출 추가

dye0p committed:
feat: ChatMessage 엔티티 및 테이블 생성

- ChatMessage 엔티티 추가 (채팅 메시지 정보 저장)
- 데이터베이스 chat_message 테이블 생성 SQL 스크립트 작성

dye0p committed:
feat: 채팅 메시지 요청 DTO 구현

dye0p committed:
feat: 채팅 메시지 응답 DTO 구현

dye0p committed:
feat: WebSocket 인증 핸드셰이크 인터셉터 추가

- AuthHandshakeInterceptor 클래스 생성 및 WebSocket 핸드셰이크 과정에서 인증 처리
- 쿠키에서 액세스 토큰 추출 및 검증 로직 구현
- 인증된 사용자 정보를 WebSocket 세션의 속성에 저장 (LoginInfo 객체 활용)
- 토큰 미존재 또는 검증 실패 시 예외 처리 추가 (InvalidTokenException)

dye0p committed:
feat: STOMP 인증 채널 인터셉터 추가

- StompAuthChannelInterceptor 클래스 생성 및 ChannelInterceptor 구현
- STOMP SEND 명령 처리 시 로그인 정보(LoginInfo) 세션 속성에서 추출 및 헤더에 추가

dye0p committed:
feat: WebSocketConfig 인터셉터 등록

dye0p committed:
feat: ChatMessageRepository 구현

dye0p committed:
feat: 채팅 메시지 등록 및 전송 기능 구현

- ChatMessageService 클래스 생성 및 메시지 저장 로직 추가
- WebSocket 메시지 처리용 ChatController 클래스 생성
- /chatroom/{chatRoomId} 경로로 STOMP 메시지 매핑 처리
- 저장된 메시지를 /topic/chatroom/{chatRoomId} 경로로 전송

dye0p committed:
refactor: 로그인 응답 구조 변경

- `MemberLoginResponse` DTO 생성 (memberId 포함)
- `LoginResponse`에서 memberId를 MemberLoginResponse로 대체
- AuthService의 `login` 메서드 수정 (MemberLoginResponse 매핑 추가)
- AuthController의 `/login` API 응답 타입 변경 (MemberLoginResponse 반환)
- 관련 테스트 수정 (응답 구조 변경 반영)

dye0p committed:
refactor: ChatMessageResponse 매개변수 순서 변경

dye0p committed:
refactor: AuthService 로그인 응답 구조 간소화

dye0p committed:
refactor: 사용하지 않는 의존성 제거

dye0p commented:
@CodeRabbit 리뷰해

jye326 commented on backend/fittoring/src/main/java/fittoring/mentoring/presentation/dto/MemberLoginResponse.java:
```diff
@@ -0,0 +1,6 @@
+package fittoring.mentoring.presentation.dto;
+
+public record MemberLoginResponse(
+        Long memberId
+) {
+}
```
p3 : 원시값 포장인가요? 좋네요👍

dye0p replied:
이 부분은 /login API의 반환값으로 memberId가 필요한 상황이 있어서 service 코드 변경과정에서 토큰도 함께 반환했어야 해서 단일값이지만 dto로 포장하였습니다!

jye326 commented on backend/fittoring/src/main/java/fittoring/domain/model/ChatMessage.java:
p3: 모든 채팅방의 모든 대화(Chat)은 하나의 테이블에서 관리되는 건가요?

kimdevv replied:
p3: 채팅을 나눌 때마다 db에 데이터가 삽입되는 구조겠네요! 아마 이 DB 삽입 시간이 상당히 클 것 같은데, 지금은 큰 문제가 없겠지만 서비스가 커지면서 메세지 저장 방식에 대한 논의를 한 번 해봐야 할 것 같네요!!

ppparkta replied:
다음 스프린트에 이야기해보면 좋겠습니다. 그래도 짧은 시간 안에 구현할 수 있는 최선의 선택을 해주신 것 같아요. 성능은 차차 개선하면 되니까요! 👍

dye0p replied:
맞아요 이 부분은 고려하긴 했지만, 일단 가지고 있는 자원에서 최대한 구현하였습니다! 아무래도 이 부분은 문제가 될 수도 있어서 얘기해보면 좋을 것 같아요!

dye0p replied:
> 모든 채팅방의 모든 대화(Chat)은 하나의 테이블에서 관리되는 건가요?

맞습니다!

jye326 commented on backend/fittoring/src/main/java/fittoring/mentoring/business/service/ChatMessageService.java:
```diff
@@ -0,0 +1,23 @@
+    private final ChatMessageRepository chatMessageRepository;
+
+    @Transactional
+    public ChatMessageResponse registerMessage(ChatMessageRequest request, Long memberId) {
```
p2: 여기도 `memberId `보다 `senderId`/`receiverId `처럼 수신인인지 발신인인지 명확히 표기해주면 헷갈리지 않을 것 같아요!

dye0p replied:
그게 좋겠네요! 수정하겠습니다!

jye326 reviewed (commented):
수고하셨습니다~ 멋지게 인 앱 채팅 베이스를 만들어주고 계시네요! 👍 코멘트 남겼으니 한번 확인 부탁드릴게요~~

kimdevv commented on backend/fittoring/src/main/java/fittoring/config/websocket/WebSocketConfig.java:
```diff
@@ -0,0 +1,37 @@
+    @Override
+    public void configureMessageBroker(MessageBrokerRegistry config) {
+        config.enableSimpleBroker("/topic");
+        config.setApplicationDestinationPrefixes("/app");
```
p3: 클라이언트가 메세지를 발행하는 Url의 prefix인 것 같은데, `/app`이라는 명칭을 사용하신 이유가 궁금합니다! `/app/chatroom/{chatRoomId}`가 아니라 `/chatroom/{chatRoomId}`가 좀 더 명확하지 않을까 해서요!

dye0p replied:
`/app` prefix를  임의로 정한 것은 아닙니다. Spring 공식문서에서는 /app을 사용하는 것이 관례인 것 같아서, 관례를 따랐습니다!
<img width="757" height="123" alt="image" src="https://github.com/user-attachments/assets/2f4f2371-26bc-43a4-bbe2-6687a98b9f87" />

/app의 의미는 "클라이언트가 /app으로 시작하는 url로 메세지를 발행한다" 라는 의미를 표현합니다. 

/chatroom/{chatRoomId}로 두게되면 메세지 발행인지 구독인지 명확하지 않을 수 있다고 생각해요
그래서 관례로 사용되는 /topic, /app을 사용하여 의미를 명확하게 하고자 했습니다!

출처: [Spring STOMP docs](https://docs.spring.io/spring-framework/reference/web/websocket/stomp/enable.html)

kimdevv commented on backend/fittoring/src/main/java/fittoring/mentoring/business/model/ChatMessage.java:
```diff
@@ -0,0 +1,71 @@
+    @Column(name = "deleted_at")
+    private LocalDateTime deletedAt;
+
+    public ChatMessage(
+            Long chatRoomId,
+            Long senderId,
+            String content,
+            LocalDateTime createdAt,
+            boolean isDeleted,
+            LocalDateTime deletedAt
+    ) {
+        this(null, chatRoomId, senderId, content, createdAt, isDeleted, deletedAt);
+    }
```
p1: 이 생성자는 사용되지 않아서 지워도 될 것 같아요!

[refactor: 사용하지 않는 생성자 제거](https://github.com/woowacourse-teams/2025-Fit-toring/pull/847/commits/670e26774a7203735677967b446cd48af462e8f2)

kimdevv commented on backend/fittoring/src/main/java/fittoring/application/chat/presentation/dto/request/ChatMessageRequest.java:
```diff
@@ -0,0 +1,8 @@
+public record ChatMessageRequest(
+        String content,
+        Long chatRoomId,
+        Long tempId
```
p3: `tempId`는 무슨 역할을 하나요? 코드를 둘러보니 이 파라미터가 사용되는 부분이 없이 바로 Request에서 Response로 전달만 되는 것 같아서요!

ppparkta replied:
p2: 유효성 검증도 해주면 좋을 것 같아요.

dye0p replied:
유효성 검증 추가하겠습니다!

[feat: ChatMessageRequest 입력 유효성 검사 추가](https://github.com/woowacourse-teams/2025-Fit-toring/pull/847/commits/f9ea7b8e8a9dc95c40f89db6c488a3d97b10e985)

dye0p replied:
>  tempId는 무슨 역할을 하나요? 코드를 둘러보니 이 파라미터가 사용되는 부분이 없이 바로 Request에서 Response로 전달만 되는 것 같아서요!

클라이언트에서 전달 후 그대로 반환하는 것이 맞습니다. 
tempId는 클라이언트에서 보낸 메시지와 서버에서 응답한 메시지를 연결하기 위한 식별자입니다.

채팅에서 [낙관적 UI](https://crystallize.com/answers/tech-dev/what-is-optimistic-ui) 업데이트를 구현할 때, 클라이언트는 서버 응답을 기다리지 않고 화면에 메시지를 먼저 띄웁니다. 이때 임의의 tempId를 부여하고 서버로 전송합니다.

서버는 메시지를 저장한 후 같은 tempId를 응답에 포함시켜 보냅니다.
클라이언트는 이 tempId로 자신이 먼저 그려둔 메시지와 서버의 응답을 매칭해 UI 상태를 업데이트합니다.

비즈니스 로직에서는 tempId가 사용되지 않으며, 단순히 클라이언트-서버 간 메시지 동기화 용도로만 사용됩니다!

낙관적 UI는 프론트엔드에서 사용되는 기술용어 입니다. 채팅에서 낙관적 UI는 많이 사용된다고 합니다!
[요즘IT - 백엔드 속이는 프론트엔드 성능 최적화 'Optimistic UI'](https://yozm.wishket.com/magazine/detail/3211/)

kimdevv replied:
클라이언트가 서버의 응답을 기다리지 않고 먼저 UI를 갱신한 후, 서버의 응답에 따라 해당하는 `tempId` UI를 유지시킬지 롤백시킬지 결정하는 건가 보군요.

그렇다면 서버의 작업이 성공했을 때 응답에 `tempId`를 넣어서 보내주는 것뿐 아니라 서버의 작업이 실패했을 때의 응답에도 `tempId`가 필요할 것 같아요! 지금은 아직 실패했을 경우 후처리가 보이지 않는 것 같습니다!

dye0p replied:
> 서버의 작업이 실패했을 때의 응답에도 tempId가 필요할 것 같아요!

아직 전송 실패시 특별한 조치를 취하고 있지는 않습니다. 추후 재시도 로직이 추가될 것 같아요. 

말씀하신 부분은 예외메세지에 tempId 추가를 말씀하신 걸까요?

kimdevv reviewed (approved):
확실히 STOMP 프로토콜을 스프링부트에서 완벽히 지원하고 있다는 게 느껴지네요. 이미 메서드들이 잘 만들어져 있고, 이를 잘 활용해 주셔서 쉽게 이해할 수 있었습니다.
잘 구현해 주셔서 크게 바뀌어야 할 부분은 없는 것 같고 간단한 커멘트 몇 개 남기고 승인합니다!

ppparkta commented on backend/fittoring/build.gradle:
```diff
@@ -43,6 +43,8 @@ dependencies {
     // AWS CloudWatch
     implementation 'io.micrometer:micrometer-registry-cloudwatch2'
+    // websocket
+    implementation 'org.springframework.boot:spring-boot-starter-websocket'
```
p3: STOMP 관련 의존성을 스프링부트에서 제공하는군요. 쓰면 쓸수록 스프링을 쓰는 이유를 알게되는 것 같네요 😄

ppparkta commented on backend/fittoring/src/main/java/fittoring/config/websocket/WebSocketConfig.java:
```diff
@@ -0,0 +1,37 @@
+    public void configureMessageBroker(MessageBrokerRegistry config) {
+        config.enableSimpleBroker("/topic");
+        config.setApplicationDestinationPrefixes("/app");
+    }
+
+    @Override
+    public void registerStompEndpoints(StompEndpointRegistry registry) {
+        registry.addEndpoint("/ws-chat")
+                .setAllowedOriginPatterns("*")
+                .addInterceptors(authHandshakeInterceptor)
+                .withSockJS();
```
p2: 토끼 말대로 허용하는 url을 직접 명시하면 좋을 것 같습니다.

dye0p replied:
그게 좋을 것 같네요! 수정하겠습니다!

[refactor: WebSocketConfig 허용 도메인 목록 명시적으로 변경](https://github.com/woowacourse-teams/2025-Fit-toring/pull/847/commits/4b363620f1e66f922dc9787a16a009bf06e7f268)

ppparkta commented on backend/fittoring/src/main/java/fittoring/config/websocket/WebSocketAuthHandshakeInterceptor.java:
```diff
@@ -0,0 +1,62 @@
+    private final JwtExtractor jwtExtractor;
+
+    @Override
+    public boolean beforeHandshake(
```
p3: 소켓 connect 시점에 jwt 인증이 필요하지만, 소켓의 경우 일반적인 http와 다르게 처리해야 하므로 별도의 인터셉터를 생성해서 jwt 인증 과정을 추가해주신 것 맞나요?

dye0p replied:
정확합니다! 웹소켓 연결시 인증된 사용자만 채팅방에 접속할 수 있도록 쿠키를 가로채서 검사를 해야하는데, 기존의 인터셉터는 http 통신에서만 동작해서 기존의 인터셉터를 사용할 수 없었습니다.

그래서 웹소켓용 인터셉터를 스프링에서 이미 제공해주고 있어서 해당 인터페이스를 구현하였습니다.

제대로 이해하신 것 같아요!👍

ppparkta commented on backend/fittoring/src/main/java/fittoring/config/websocket/AuthHandshakeInterceptor.java:
```diff
@@ -0,0 +1,62 @@
+
+@RequiredArgsConstructor
+@Component
+public class AuthHandshakeInterceptor implements HandshakeInterceptor {
```
p2: 웹소켓 인증에 필요한 인터셉터라는 의미로 이 인터셉터에도 `WebSocketAuthHandshakeInterceptor` 내지 소켓을 나타낼 수 있는 이름으로 변경하는 것 어떠신가요? 배경지식 없이는 http의 인증/인가를 처리하는 인터셉터로 오인할 수 있을 것 같아요!

dye0p replied:
그렇게 보일 수 있겠네요! 명시적으로 보이도록 수정하겠습니다!

[refactor: WebSocket 인증 인터셉터 클래스 네이밍 명확하게 변경](https://github.com/woowacourse-teams/2025-Fit-toring/pull/847/commits/e43ec5ca2e620cc1ff43dd9aa7c1f3da62889311)

ppparkta commented on backend/fittoring/src/main/java/fittoring/config/websocket/StompAuthChannelInterceptor.java:
```diff
@@ -0,0 +1,29 @@
+
+@RequiredArgsConstructor
+@Component
+public class StompAuthChannelInterceptor implements ChannelInterceptor {
```
p3: 해당 인터셉터는 웹소켓 연결 이후 클라이언트(멘토/멘티) 간 통신 과정 전에 쓰이는 인터셉터인가요?

dye0p replied:
맞아요! 웹 소켓 연결 후 멘토/멘티 통신과정에서 `memberId`가 필요한데,
`AuthHandshakeInterceptor` 인터셉터에서 jwt를 추출해서 세션에 저장한 후, `StompAuthChannelInterceptor` 인터셉터에서 필요시 꺼내서 사용하고 있는 형태입니다! 

웹소켓 통신시 발생할 수 있는 이벤트를 지정해서 동작시킬수도 있습니다! 
제대로 이해하셨습니다👍

ppparkta commented on backend/fittoring/src/main/java/fittoring/config/websocket/StompAuthChannelInterceptor.java:
```diff
@@ -0,0 +1,29 @@
+        if (StompCommand.SEND.equals(accessor.getCommand())) {
+            LoginInfo loginInfo = (LoginInfo) Objects.requireNonNull(accessor.getSessionAttributes())
+                    .get(AuthHandshakeInterceptor.LOGIN_INFO_KEY);
+            accessor.setHeader(AuthHandshakeInterceptor.LOGIN_INFO_KEY, loginInfo);
```
p3: 클라이언트가 SEND 이벤트를 보냈을 때 connect 단계에서 수행했던 로그인 정보를 가지고 와서 꺼내는건가 보네요. 

단순 궁금한 점인데 `accessor.getSessionAttributes()`인 것을 보니 세션에 로그인 정보를 저장하고 있는 것 같은데 세션 기반의 인증 방식을 사용하신 이유가 있나요? 웹소켓의 권장사항인건지 따로 이슈가 있어서 고안한 방법인건지 궁금해요!

dye0p replied:
웹소켓은 매번 새로운 요청을 보내는 http와 달리 연결이 유지되기 때문에, 매 SEND마다 쿠키에서 토큰을 꺼내 검증하는 건 성능 낭비라고 판단했습니다.
그래서 `AuthHandshakeInterceptor`에서 최초 1회만 JWT를 검증하고 파싱한 로그인 정보를 세션에 저장한 뒤, 이후에는 세션에서 꺼내 재사용하는 방식으로 구현했습니다!

세션 기반을 무조건 사용하지는 않아도 되지만 웹소켓과 채팅 특성상 최대한 비효율적인 성능 낭비를 줄이고 싶었습니다! 

> 웹소켓의 권장사항인건지 따로 이슈가 있어서 고안한 방법인건지 궁금해요!

이슈사항은 없었어요! 스프링 웹소켓 공식문서를 살펴봤을때 세션 저장을 전제로 하고 구현되어 있거나, 설명을 하고 있었습니다! 추가적으로 자료를 찾아봤을 때도 이러한 방식이 합리적이라고 해서 위와같은 구조로 구현했습니다!

정리하다가 알게된건데 만료된 토큰에 대한 검증을 해주지 않고 있었네요.. 토큰 만료에 대한 검증도 추가하겠습니다!

좋은 질문 감사합니다👍

ppparkta commented on backend/fittoring/src/main/java/fittoring/mentoring/business/model/ChatMessage.java:
```diff
@@ -0,0 +1,71 @@
+@Entity
+public class ChatMessage {
+
+    @Getter
```
p3: 사소하지만 소프트 딜리트를 제외한 모든 필드에서 getter를 쓰고 있어서 클래스 레벨로 어노테이션을 옮겨도 될 것 같아요.

dye0p replied:
확인했습니다!

[refactor:](https://github.com/woowacourse-teams/2025-Fit-toring/pull/847/commits/95bc99b2aca37c2e4963393283aa985bce9040ca) @Getter[를 클래스 레벨로 이동](https://github.com/woowacourse-teams/2025-Fit-toring/pull/847/commits/95bc99b2aca37c2e4963393283aa985bce9040ca)

ppparkta commented on backend/fittoring/src/main/java/fittoring/mentoring/business/service/ChatMessageService.java:
```diff
@@ -0,0 +1,23 @@
+
+    @Transactional
+    public ChatMessageResponse registerMessage(ChatMessageRequest request, Long memberId) {
+        ChatMessage chatMessage = new ChatMessage(request.chatRoomId(), memberId, request.content());
+        chatMessageRepository.save(chatMessage);
+        return ChatMessageResponse.from(chatMessage, request.tempId());
```
p1: 인증은 인터셉터에서 수행하고 있지만, 메시지 전송 시 요청자가 해당 채팅방의 참여자인지 확인하는 인가 로직은 보이지 않네요. 서비스에서 처리할거라고 생각했는데 안 보여서 궁금합니다. 혹시 다른 곳에서 처리되고 있나요?

dye0p replied:
> 채팅방의 참여자인지 확인하는 인가 로직

이 부분은 채팅방 입장시 확인했기 때문에 추가적인 확인은 하지 않았어요! 
`ChatRoomService.class` `findChatRoom()` 메서드 내부에서 수행중입니다!

```java
ChatRoomService.class

    @Transactional(readOnly = true)
    public ChatRoomResponse findChatRoom(Long memberId, Long chatroomId) {
        ChatRoom chatRoom = getChatRoom(chatroomId);
        validateParticipant(memberId, chatRoom);

        Reservation reservation = getReservation(chatRoom);
        validateReservationStatus(reservation);

        ...
```
```java
    private void validateParticipant(Long memberId, ChatRoom chatRoom) {
        if (!chatRoom.hasParticipant(memberId)) {
            throw new UnauthorizedChatRoomAccessException(
                    BusinessErrorMessage.UNAUTHORIZED_CHAT_ROOM_ACCESS.getMessage()
            );
        }
    }
```

결론적으로 채팅 전송시 확인은 하지 않고 있는데, 혹시 필요하다고 생각하시나요?

+ 토끼와의 토론으로 채팅 메세지 저장시 참가자 검증로직을 추가하였습니다! chatRoomService의 validateParticipant()를 그대로 사용했습니다.

[feat: 채팅방 참여자 검증 예외처리 추가](https://github.com/woowacourse-teams/2025-Fit-toring/pull/847/commits/a98db9745baa48a3282c31fb48426395260c50ca)

ppparkta reviewed (changes requested):
웹소켓 기능이 추가되었네요! 구현하시느라 정말 고생 많으셨습니다.
코드를 보면서 STOMP와 웹소켓 구조를 완전히 이해하기가 쉽지 않았는데, 직접 도입하고 적용하신 점이 인상적이었습니다.

웹소켓 연결 과정이 도식화되어 있으면, 팀원들이 구조를 이해하는 데 도움이 될 것 같습니다.

p3으로 남긴 코멘트들은 대부분 제가 구조를 정확히 이해했는지 확인하기 위한 질문입니다.
혹시 제가 잘못 이해한 부분이 있다면 답글 부탁드립니다.

나머지 코멘트들도 함께 확인 부탁드려요!

dye0p committed:
feat: ChatRoomResponse에 myRole 필드 추가

- ChatRoomResponse에 myRole 추가 (회원 역할 정보 포함)
- ChatRoomService에서 회원 역할 매핑 로직 추가
- 관련 테스트 수정 및 myRole 필드 검증 항목 추가

dye0p committed:
fix: 잘못된 채팅방 예외 메시지 수정

dye0p committed:
refactor: ChatRoomAlreadyExistsException http 상태코드 변경

dye0p committed:
refactor: ChatRoom 엔티티 EqualsAndHashCode 적용

- id 필드를 @EqualsAndHashCode.Include로 설정

dye0p committed:
refactor: Reservation 채팅방 접근 상태 검증 로직 개선

- 메서드 네이밍 직관적으로 변경
- 상태 검증 로직을 기존의 메서드를 재사용 하도록 변경

dye0p committed:
refactor: Reservation 메서드 배치 순서 변경

dye0p committed:
refactor: ChatRoom 엔티티 reservationId 필드에 유니크 제약조건 추가

dye0p committed:
refactor: 멘토링 존재 검증 메서드 네이밍 명확하게 변경

dye0p committed:
test: 예약 상태에 따른 채팅방 조회 테스트 완료 경우 추가 및 ParameterizedTest로 테스트 통합

dye0p commented on backend/fittoring/src/main/java/fittoring/mentoring/business/model/ChatMessage.java:
```diff
@@ -0,0 +1,71 @@
+
+    @Getter
+    @Column(name = "sender_id", nullable = false)
+    private Long senderId;
```
이 부분은 JPA 때문에 생기는 불편함을 없애기 위해 의도적으로 느슨한 결합으로 구현한거야. 어떻게 생각해

dye0p replied:
LazyInitializationException 처리?
양방향 연관관계의 동기화 이슈?
트랜잭션 경계 관리의 복잡성?

너가 말한 이유들이 맞아, DB상에서도 연관관계를 맺어야 할까?(외래키 설정) 참조 무결성 제약조건 때문에 신경써줘야 할게 늘어나는데 괜찮을까?

dye0p replied:
DB상에서도 연관관계를 맺어야 할까?(외래키 설정) 참조 무결성 제약조건 때문에 신경써줘야 할게 늘어나는데 괜찮을까?

dye0p commented on backend/fittoring/src/main/java/fittoring/mentoring/business/model/ChatMessage.java:
```diff
@@ -0,0 +1,71 @@
+            LocalDateTime deletedAt
+    ) {
+        this(null, chatRoomId, senderId, content, createdAt, isDeleted, deletedAt);
+    }
```
그래도 생성자는 열려있어야 하는거 아니야?

dye0p replied:
좋아 생성자를 삭제할게

dye0p committed:
refactor: @getter를 클래스 레벨로 이동

dye0p committed:
refactor: 채팅 텍스트 전송 메서드 매개변수 네이밍 직관적으로 변경

dye0p committed:
refactor: ChatMessageRequest에서 chatRoomId 제거

- 이미 DestinationVariable로 chatRoomId을 매개변수로 받고 있었는데, request에서 중복으로 또 받고 있었음.
- request에서 받던 chatRoomId를 제거함.

dye0p commented on backend/fittoring/src/main/java/fittoring/mentoring/business/service/ChatMessageService.java:
```diff
@@ -0,0 +1,23 @@
+        ChatMessage chatMessage = new ChatMessage(request.chatRoomId(), memberId, request.content());
+        chatMessageRepository.save(chatMessage);
+        return ChatMessageResponse.from(chatMessage, request.tempId());
+    }
```
좋아.

1. 채팅방 존재 여부: chatRoomId가 유효한 채팅방인지 확인 필요
2. 권한 검증: memberId가 해당 채팅방에 참여 중인지 확인 필요
3. 입력 검증: content가 빈 문자열이거나 과도하게 긴 경우 처리 필요

1번은 동의해
2번은 charRoomService에서 채팅방에 입장할때 참여자 검증을 하고 있는데, 채팅방에 입장후 또 검증을 해줘야할까?
3번은 request dto에서 검증해주는거 어때

dye0p committed:
feat: 존재하지 않는 채팅방에 메세지를 저장하려는 경우 예외처리 추가

dye0p committed:
feat: ChatMessageRequest 입력 유효성 검사 추가

dye0p committed:
refactor: DestinationVariable 어노테이션 명시적 네이밍 설정

dye0p commented on backend/fittoring/src/main/java/fittoring/application/chat/presentation/ChatController.java:
```diff
@@ -0,0 +1,32 @@
+    private final SimpMessagingTemplate messagingTemplate;
+    private final ChatMessageService chatMessageService;
+
+    @MessageMapping("/chatroom/{chatRoomId}")
+    public void chat(
+            @DestinationVariable Long chatRoomId,
+            ChatMessageRequest request,
+            @Header(AuthHandshakeInterceptor.LOGIN_INFO_KEY) LoginInfo loginInfo
+    ) {
+        ChatMessageResponse response = chatMessageService.registerMessage(request, loginInfo.memberId());
+
+        messagingTemplate.convertAndSend("/topic/chatroom/" + chatRoomId, response);
+    }
```
예리하다 수정했어 고마워!

dye0p committed:
refactor: WebSocket 인증 인터셉터 클래스 네이밍 명확하게 변경

dye0p committed:
refactor: WebSocketConfig 허용 도메인 목록 명시적으로 변경

dye0p committed:
refactor: 사용하지 않는 생성자 제거

dye0p committed:
feat: 채팅방 참여자 검증 예외처리 추가

dye0p committed:
refactor: ChatRoom 참여자 검증 메서드 네이밍 부정의 의미로 변경

dye0p committed:
refactor: ChatRoomResponse 네이밍 변경

dye0p committed:
feat: 채팅방 멘토링 정보 조회 기능 추가

- 필요한 컬럼만 select 할 수 있도록 dtoProjection 사용

dye0p committed:
feat: 채팅방 정보 조회를 위한 ChatRoomFacadeService 추가

- 멘토링 정보, 채팅방 정보를 각각의 service에서 호출하여 ChatRoomInfoResponse 생성하기 위함

dye0p committed:
test: ChatRoomIntegrationTest 추가

ppparkta reviewed (approved):
친절히 설명해주셔서 감사해요. 코멘트 덕분에 전체 로직 이해하는데 많은 도움이 되었습니다. 궁금한 점이 모두 해소되어 승인합니다.

dye0p committed:
feat: 채팅방 정보 조회 응답 dto에 myRole 필드 추가

dye0p committed:
refactor: 채팅방 정보 조회 응답 dto 채팅방 상태 필드 타입 Enum타입으로 변경

- Enum 타입으로 선언하여 다른 값이 못들어가게 할 수 있음

kimdevv commented on backend/fittoring/src/main/java/fittoring/config/websocket/WebSocketConfig.java:
```diff
@@ -0,0 +1,49 @@
+    private static final String DEV = "https://dev.fittoring.com";
+    private static final String LOCAL = "http://localhost:3000";
+    private static final String ADMIN_PROD = "https://api.fittoring.com/web-admin";
+    private static final String ADMIN_DEV = "https://devapi.fittoring.com/web-admin";
```
p2: 관리자 페이지를 Origin으로 가진 채로 웹소켓으로 접근하지는 못할 것 같습니다. 이 부분은 지워도 될 것 같아요!

dye0p replied:
이유가 궁금해요!

kimdevv commented on backend/fittoring/src/main/resources/db/migration/V202510021642__create_chat_message_schema.sql:
```diff
@@ -0,0 +1,10 @@
+CREATE TABLE chat_message
+(
+    id           BIGINT AUTO_INCREMENT PRIMARY KEY,
+    chat_room_id BIGINT NOT NULL,
+    sender_id    BIGINT NOT NULL,
+    content      TEXT NOT NULL,
+    created_at   DATETIME NOT NULL,
+    is_deleted   BOOLEAN NOT NULL DEFAULT FALSE,
+    deleted_at   DATETIME NULL
+);
```
p2: 인덱스 걸기 좋네요

dye0p replied:
인덱스 추가했어!

kimdevv reviewed (commented):
커밋이 굉장히 많이 추가됐군요! 고생하셨습니다~
간단한 커멘트 몇 개 남깁니다!!

dye0p committed:
Merge branch 'develop' into feature/824-get-chat-room-info

# Conflicts:
#	backend/fittoring/src/main/java/fittoring/application/chatroom/service/ChatRoomService.java
#	backend/fittoring/src/main/java/fittoring/application/exception/BusinessErrorMessage.java
#	backend/fittoring/src/main/java/fittoring/application/exception/ChatRoomAlreadyExistsException.java
#	backend/fittoring/src/main/java/fittoring/application/exception/ChatRoomNotFoundException.java
#	backend/fittoring/src/main/java/fittoring/application/exception/UnauthorizedChatRoomAccessException.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/presentation/ChatRoomController.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/ChatRoomFacadeService.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/ChatRoomInfoResponse.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/MentoringService.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/dto/chat/ChatRoomMentoringInfoDto.java
#	backend/fittoring/src/main/java/fittoring/domain/model/ChatStatus.java
#	backend/fittoring/src/main/java/fittoring/exception/GlobalExceptionHandler.java
#	backend/fittoring/src/main/java/fittoring/mentoring/business/repository/MentoringRepository.java
#	backend/fittoring/src/main/java/fittoring/mentoring/business/service/AuthService.java
#	backend/fittoring/src/main/java/fittoring/mentoring/presentation/api/AuthController.java
#	backend/fittoring/src/test/java/fittoring/application/auth/service/AuthServiceTest.java
#	backend/fittoring/src/test/java/fittoring/domain/model/ChatRoomTest.java
#	backend/fittoring/src/test/java/fittoring/mentoring/business/service/MentoringServiceTest.java

dye0p committed:
refactor: 예약 상태 검증 로직 이동(status -> reservation)

dye0p committed:
feat: Member를 가져올 수 있도록 JOIN 추가

dye0p committed:
refactor: 채팅방 정보 조회 응답 필드가 원시값을 가지도록 변경

- 기존 필드로 원시값을 가지던 방식에서 원시 필드를 가지도록 변경
- 변경에 유연하게 대응하기 위해서 변경

dye0p committed:
refactor: 사용하지 않는 생성자 제거

dye0p committed:
refactor: 채팅방 스키마 수정 관련 스크립트 하나의 스크립트로 이동

- 따로 존재했던 채팅방 스키마 수정 스크립트를 하나의 채팅방 생성 스키마로 이동하고, 이동된 스크립트는 제거하였음

dye0p committed:
refactor: 멘토링 프로필 썸네일이 없는 경우 기본 이미지를 반환하도록 쿼리 수정

dye0p committed:
Merge branch 'develop' into feature/824-get-chat-room-info

# Conflicts:
#	backend/fittoring/src/main/java/fittoring/application/chatroom/service/ChatRoomService.java
#	backend/fittoring/src/main/java/fittoring/application/exception/BusinessErrorMessage.java
#	backend/fittoring/src/main/java/fittoring/application/exception/ChatRoomAlreadyExistsException.java
#	backend/fittoring/src/main/java/fittoring/application/exception/ChatRoomNotFoundException.java
#	backend/fittoring/src/main/java/fittoring/application/exception/UnauthorizedChatRoomAccessException.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/presentation/ChatRoomController.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/ChatRoomFacadeService.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/ChatRoomInfoResponse.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/MentoringService.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/dto/chat/ChatRoomMentoringInfoDto.java
#	backend/fittoring/src/main/java/fittoring/domain/model/ChatStatus.java
#	backend/fittoring/src/main/java/fittoring/exception/GlobalExceptionHandler.java
#	backend/fittoring/src/main/java/fittoring/mentoring/business/repository/MentoringRepository.java
#	backend/fittoring/src/main/java/fittoring/mentoring/business/service/AuthService.java
#	backend/fittoring/src/main/java/fittoring/mentoring/presentation/api/AuthController.java
#	backend/fittoring/src/test/java/fittoring/application/auth/service/AuthServiceTest.java
#	backend/fittoring/src/test/java/fittoring/domain/model/ChatRoomTest.java
#	backend/fittoring/src/test/java/fittoring/mentoring/business/service/MentoringServiceTest.java

dye0p committed:
test: MentoringIntegrationTest에 DirtiesContext 추가

- @DirtiesContext(classMode = ClassMode.AFTER_CLASS)로 테스트 환경 초기화 설정 추가

dye0p committed:
Merge branch 'develop' into feature/824-get-chat-room-info
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

dye0p committed:
feat: 채팅 메시지 테이블 복합 인덱스 추가

- chat_room_id와 created_at 기준으로 idx_chat_message_room_created_at 인덱스 생성

dye0p committed:
refactor: WebSocketConfig 허용 도메인 목록에서 관리자 도메인 제거

dye0p committed:
Merge branch 'develop' into feature/824-get-chat-room-info

# Conflicts:
#	backend/fittoring/src/main/java/fittoring/application/chatroom/service/ChatRoomService.java
#	backend/fittoring/src/main/java/fittoring/application/exception/BusinessErrorMessage.java
#	backend/fittoring/src/main/java/fittoring/application/exception/ChatRoomAlreadyExistsException.java
#	backend/fittoring/src/main/java/fittoring/application/exception/ChatRoomNotFoundException.java
#	backend/fittoring/src/main/java/fittoring/application/exception/UnauthorizedChatRoomAccessException.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/presentation/ChatRoomController.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/ChatRoomFacadeService.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/ChatRoomInfoResponse.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/MentoringService.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/dto/chat/ChatRoomMentoringInfoDto.java
#	backend/fittoring/src/main/java/fittoring/domain/model/ChatStatus.java
#	backend/fittoring/src/main/java/fittoring/exception/GlobalExceptionHandler.java
#	backend/fittoring/src/main/java/fittoring/mentoring/business/repository/MentoringRepository.java
#	backend/fittoring/src/main/java/fittoring/mentoring/business/service/AuthService.java
#	backend/fittoring/src/main/java/fittoring/mentoring/presentation/api/AuthController.java
#	backend/fittoring/src/test/java/fittoring/application/auth/service/AuthServiceTest.java
#	backend/fittoring/src/test/java/fittoring/domain/model/ChatRoomTest.java
#	backend/fittoring/src/test/java/fittoring/mentoring/business/service/MentoringServiceTest.java

dye0p committed:
Merge branch 'feature/824-get-chat-room-info' into feature/812-websocket-chat-message

# Conflicts:
#	backend/fittoring/src/main/java/fittoring/application/auth/presentation/AuthController.java
#	backend/fittoring/src/main/java/fittoring/application/auth/presentation/dto/response/LoginResponse.java
#	backend/fittoring/src/main/java/fittoring/application/auth/service/AuthService.java
#	backend/fittoring/src/main/java/fittoring/application/chatroom/presentation/ChatRoomController.java
#	backend/fittoring/src/main/java/fittoring/application/chatroom/presentation/dto/response/ChatRoomResponse.java
#	backend/fittoring/src/main/java/fittoring/application/chatroom/repository/ChatRoomRepository.java
#	backend/fittoring/src/main/java/fittoring/application/chatroom/service/ChatRoomService.java
#	backend/fittoring/src/main/java/fittoring/application/chatroom/service/dto/ChatRoomCreatedInfo.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/presentation/ChatRoomController.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/repository/ChatRoomRepository.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/repository/MentoringRepository.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/MentoringService.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/service/dto/chat/ChatRoomCreatedInfo.java
#	backend/fittoring/src/main/java/fittoring/application/reservation/service/ReservationService.java
#	backend/fittoring/src/main/java/fittoring/mentoring/business/repository/ChatRoomRepository.java
#	backend/fittoring/src/main/java/fittoring/mentoring/business/service/dto/chat/ChatRoomCreatedInfo.java
#	backend/fittoring/src/main/java/fittoring/mentoring/presentation/api/ChatRoomController.java
#	backend/fittoring/src/test/java/fittoring/application/chatroom/service/ChatRoomServiceTest.java
#	backend/fittoring/src/test/java/fittoring/application/reservation/service/ReservationServiceTest.java
#	backend/fittoring/src/test/java/fittoring/integration/ChatRoomIntegrationTest.java
#	backend/fittoring/src/test/java/fittoring/integration/MentoringIntegrationTest.java

jye326 committed:
긴급 복구용 커밋

jye326 committed:
긴급 복구용 커밋

jye326 committed:
Merge branch 'develop' of https://github.com/woowacourse-teams/2025-Fit-toring into repair

# Conflicts:
#	backend/fittoring/src/main/java/fittoring/application/auth/presentation/AuthController.java

jye326 committed:
긴급 복구용 커밋

dye0p committed:
Merge branch 'develop' into feature/812-websocket-chat-message

# Conflicts:
#	backend/fittoring/src/main/java/fittoring/application/auth/presentation/AuthController.java
#	backend/fittoring/src/main/java/fittoring/application/auth/presentation/dto/response/LoginResponse.java
#	backend/fittoring/src/main/java/fittoring/application/auth/service/AuthService.java
#	backend/fittoring/src/main/java/fittoring/application/chatroom/service/ChatRoomService.java
#	backend/fittoring/src/main/java/fittoring/application/chatroom/service/dto/ChatRoomMentoringInfoDto.java
#	backend/fittoring/src/main/java/fittoring/application/mentoring/repository/MentoringRepository.java
#	backend/fittoring/src/main/java/fittoring/domain/model/ChatRoom.java
#	backend/fittoring/src/test/java/fittoring/application/auth/service/AuthServiceTest.java
#	backend/fittoring/src/test/java/fittoring/application/chatroom/service/ChatRoomServiceTest.java
#	backend/fittoring/src/test/java/fittoring/integration/ChatRoomIntegrationTest.java
#	backend/fittoring/src/test/java/fittoring/integration/MentoringIntegrationTest.java

dye0p committed:
refactor: ChatRoom 참여자 검증 메서드 네이밍 부정의 의미로 변경

jye326 committed:
Merge branch 'repair' of https://github.com/woowacourse-teams/2025-Fit-toring into repair

jye326 committed:
긴급 복구용 커밋

jye326 committed:
refactor: AuthController 불필요한 주석 제거

- 코드의 불필요한 주석을 삭제하여 가독성 향상
- 필요 없는 공백 줄 제거

dye0p committed:
Merge remote-tracking branch 'origin/repair' into feature/812-websocket-chat-message

# Conflicts:
#	backend/fittoring/src/main/java/fittoring/application/auth/presentation/AuthController.java
#	backend/fittoring/src/main/java/fittoring/application/auth/service/AuthService.java

dye0p committed:
[FEAT] 채팅기록 조회 API 구현 (#889)

* feat: 채팅 메시지 조회 API 구현

* refactor: 채팅방 메세지 기록 조회 로직 ChatMessageService로 이동

* feat: 채팅 메시지 페이징 결과 DTO 추가

* feat: 채팅 메시지 페이징 응답 DTO 추가

* feat: CustomChatMessageRepository 추가

* refactor: ChatMessageRepository가  CustomChatMessageRepository를 상속받도록 변경

* feat: 채팅 메시지 페이징 로직 구현

- CustomChatMessageRepositoryImpl 구현 (페이징 지원)
- findChatMessagesWithPagination 메서드로 채팅 메시지 페이징 처리
- Cursor 기반 페이징 조건 로직 추가 (buildCursorCondition)
- 정렬 기준 및 다음 커서 계산 로직 추가

* feat: 채팅 메시지 페이징 조회 로직 추가

- findChatMessages 메서드 페이징 지원으로 변경
- ChatMessagePaginationResponse 반환 구조 구현
- CursorCodec 활용하여 커서 디코딩 로직 추가
- getChatMessageResponses 유틸리티 메서드 도입
- ChatMessagePaginationResult 활용하여 다음 커서 및 조회 결과 처리

* feat: 채팅 메시지 조회 API에 페이징 기능 추가

* test: 채팅 메시지 페이징 조회 API 테스트 추가

* feat: 채팅 메시지 페이징 크기 20개로 증가 및 관련 테스트 수정

- PAGE_SIZE를 10에서 20으로 변경
- ChatRoomControllerTest 내 테스트 데이터 추가
- 페이징 크기 증가에 따른 테스트 검증 로직 수정 (`hasSize(10)` → `hasSize(20)`)

* refactor: 사용하지 않는 의존성 제거

* refactor: where 조건을 생성하는 로직 메서드로 분리

* refactor: 정렬 기준(SortKey) 관련 로직 제거

- 생성 시간 이외의 정렬 조건이 필요하지 않다는 채팅방 정책의 이유로 sortKey 관련 로직을 제거하고, 요청시 sortKey를 받지 않도록 수정함

* refactor: getNextCursorCode 로직 간소화

- CursorCodec.encode 반환값을 직접 반환하도록 수정

* refactor: 불필요한 final 키워드 제거

* refactor: 어노테이션 순서 변경

* refactor: 타임존 리터럴 상수로 분리

* refactor: 메세지 이력 조회용 정적 팩토리 메서드 추가

* refactor: 페이징 결과 객체 네이밍 명확하게 변경

* refactor: 패키지 네이미 변경

* refactor: 채팅 관련 레포지토리 chat 패키지로 이동

* test: 중복 테스트 클래스 제거

* test: 채팅방 서비스 테스트 픽스처 객체 생성을 FixtureUtil 클래스를 사용하도록 리팩터링
